File: forge/src/main/java/juuxel/adorn/platform/forge/compat/EcologicsCompat.java
package juuxel.adorn.platform.forge.compat;

import juuxel.adorn.block.variant.BlockVariant;
import juuxel.adorn.block.variant.PrefixedBlockVariantSet;

import java.util.List;

public final class EcologicsCompat extends PrefixedBlockVariantSet {
    @Override
    public String getModId() {
        return "ecologics";
    }

    @Override
    public List<BlockVariant> getWoodVariants() {
        return createVariants(
            BlockVariant.Wood::new,
            "azalea",
            "flowering_azalea",
            "coconut",
            "walnut"
        );
    }
}

File: forge/src/main/kotlin/juuxel/adorn/platform/forge/compat/EcologicsCompat.kt
package juuxel.adorn.platform.forge.compat

import juuxel.adorn.block.variant.BlockVariant
import juuxel.adorn.block.variant.BlockVariantSet

object EcologicsCompat : BlockVariantSet {
    override val woodVariants = listOf(
        "azalea",
        "flowering_azalea",
        "coconut",
        "walnut",
    ).map { BlockVariant.Wood("ecologics/$it") }
}
